src/video_utils.py

```diff
@@ -164,13 +166,75 @@ def get_video_duration(video_path):
         return None
 
 
+def download_instagram_images(url, temp_dir):
+    """
+    Downloads Instagram media using gallery-dl.
+
+    This function uses the `gallery-dl` command-line tool to download media from Instagram.
+    The media is stored in a temporary directory. The function returns the path to the downloaded
+    media file if successful.
+
+    Parameters:
+        url (str): The URL of the Instagram media to download.
+        temp_dir (str): The path to the temporary directory to store the downloaded media.
+
+    Returns:
+        str: The path to the downloaded media file if successful, or None if the download fails.
+    """
+    result_path = None  # Initialize the result variable
+
+    # Validate Instagram URL
+    if not re.match(r'^https?://(?:www\.)?instagram\.com/[^/]+/[^/]+/?.*$', url):
+        error("Invalid Instagram URL")
+        return None
+
+    # Ensure that reels not in the URL
+    if "reel" in url:
+        error("Reels should be handled by the yt-dlp")
+        return None
+
+    command = [
+        "gallery-dl",  # Assuming gallery-dl is installed and in the PATH
+        *(["--cookies", "instagram_cookies.txt"] if INSTACOOKIES else []),
+        url,
+        "-d",
+        temp_dir,
+    ]
+
+    try:
+        debug("Running gallery-dl command: %s", command)
+        subprocess.run(command, check=True, timeout=120)
+        result_path = []  # Initialize the result variable as a empty list
+        # Use Path.rglob to recursively search for files in the temp directory
+        # as the output contains subdirectories and may contain multiple files
+        for file in [str(file) for file in Path(temp_dir).rglob("*")]:
+            if file.endswith((".mp4", ".jpg", ".jpeg", ".png")):
```

Comment:
Does `gallery-dl` also works with .mp4 files? if so, why function name is `download_instagram_images` ?

Reply:
> Does `gallery-dl` also works with .mp4 files? if so, why function name is `download_instagram_images` ?

it should be renamed to `download_instagram_media`

Reply:
renamed